keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -1189,9 +1189,54 @@ based on storage capacity:
   to available storage and thus could run on a new node, the
```

Comment:
I don't think network-attached storage should be much of a problem.

When we are scaling up a nodepool or a nodegroup, we know in which topology it would end up. So we can query the `CSIStorageCapacity` that matches.

Reply:
For scale up from zero, network-attached storage has a similar problem: because the CSI driver hasn't provided topology information yet for the non-existent nodes, the cluster admin must set labels manually in the configuration of the node pool.

Reply:
I guess this is for the case where the nodepool is a new topology domain.

I think it should be responsibility of the CSI driver to setup that up even if no nodes exist. It's network-attached after all. It should come from somewhere and the driver should know about it. No?

Reply:
But how does the CSI driver know that the information is needed? Suppose the storage backend has many different topology segments, of which none is currently used by a specific Kubernetes cluster. Should the CSI driver report about all of them?

There's also a practical problem here: the CSI spec doesn't have support for listing topology segments for nodes where no CSI driver is running. If we want something like this, we need a new API.

Reply:
> Suppose the storage backend has many different topology segments, of which none is currently used by a specific Kubernetes cluster. Should the CSI driver report about all of them?

Sure, why not?  It may not work in every case but it seems like it could work in some?

Reply:
> Sure, why not?

Scalability and performance issues? It'll depend on the number of those additional segments, of course.

> It may not work in every case but it seems like it could work in some?

Yes. I just wanted to point out some of the details (conceptual and practical) that'll have to be addressed for this to work. I agree that it can work.